pages/en/lb3/Discovering-models-from-relational-databases.md

```diff
@@ -23,69 +23,182 @@ This process is called _discovery_ and is supported by the following connectors:
 ```
 
+Node.js does not allow the main module function to be async,
+the top-level code cannot use `await` keyword for flow control.
+As a workaround, we need to wrap the discovery code in a helper function
+that's executed using Promise API.
+
+```js
+discover().then(
+  success => process.exit(),
+  error => { console.error('UNHANDLED ERROR:\n', error); process.exit(1); },
+);
+```
+
+With the preparations done, it's time to roll up our sleeves
+and do the actual model discovery.
+
+```js
+async function discover() {
+  // It's important to pass the same "options" object to all calls
+  // of dataSource.discoverSchemas(), it allows the method to cache
+  // discovered related models
+  const options = {relations: true};
+
+  // Discover models and relations
+  const salariesSchemas = await db.discoverSchemas('salaries', options);
+  const employeesSchemas = await db.discoverSchemas('employees', options);
+
+  // Create model definition files
+  await mkdirp('common/models');
+  await writeFile(
+    'common/models/salaries.json',
+    JSON.stringify(salariesSchemas['employees.salaries'], null, 2)
+  );
+  await writeFile(
+    'common/models/employees.json',
+    JSON.stringify(salariesSchemas['employees.employees'], null, 2)
+  );
+
+  // Expose models via REST API
+  const configJson = await readFile('server/model-config.json', 'utf-8');
+  console.log('MODEL CONFIG', configJson);
+  const config = JSON.parse(configJson);
+  config.Salaries = {dataSource: DATASOURCE_NAME, public: true};
+  config.Employees = {dataSource: DATASOURCE_NAME, public: true};
+  await writeFile(
+    'server/model-config.json',
+    JSON.stringify(config, null, 2)
+  );
+}
+```
+
+Next steps:
+
+ 1. Run the discovery script to add the discovered models to your project
+    (or update the existing definitions).
+ 2. Start your application in the usual way, e.g. via `npm start` or `node .`
+ 3. Use [API Explorer](./Use-API-Explorer.md) to inspect your newly defined
+    REST API.
+ 4. Notice that "salaries" model exposes BelongsTo relation with "employees"
+    model. There is no counter relation "employees has many salaries",
+    you have to define this relation yourself.
+
 ## Additional discovery functions
 
 Some connectors provide discovery capability so that we can use DataSource to discover model definitions from existing database schema.
 The following APIs enable UI or code to discover database schema definitions that can be used to build LoopBack models.
 
-`discoverModelDefinitions()` by default discovers database schema definitions owned by the userid used to connect to the database.  
+`discoverModelDefinitions()` by default discovers database schema definitions owned by the userid used to connect to the database.
 Calling it with the `all: true` option makes the discovery include the database schema definitions of all owners.
 
 {% include code-caption.html content="/server/bin/script.js" %}
```

Comment:
Maybe (for consistency with the other examples):
`PRODUCT` => `employees`
`INVENTORY` => `salaries`

Reply:
Ah, good catch. I'll do the opposite change and update my new examples to use `PRODUCT` and `INVENTORY`.